src/Categories/Functor/Bifunctor.agda

```diff
@@ -11,23 +12,60 @@ open import Categories.Category.Product
 
 private
   variable
-    o ℓ e o′ ℓ′ e′ o″ ℓ″ e″ : Level
-    C D D₁ D₂ E E₁ E₂ : Category o ℓ e
+    o ℓ e o′ ℓ′ e′ o″ ℓ″ e″ o‴ ℓ‴ e‴ o⁗ ℓ⁗ e⁗ : Level
+    C : Category o ℓ e
+    D D₁ D₂ : Category o′ ℓ′ e′
+    E E₁ E₂ : Category o″ ℓ″ e″
+    A A₁ A₂ : Category o‴ ℓ‴ e‴
+    B B₁ B₂ : Category o⁗ ℓ⁗ e⁗
 
 Bifunctor : Category o ℓ e → Category o′ ℓ′ e′ → Category o″ ℓ″ e″ → Set _
 Bifunctor C D E = Functor (Product C D) E
 
-overlap-× : ∀ (H : Bifunctor D₁ D₂ C) (F : Functor E D₁) (G : Functor E D₂) → Functor E C
-overlap-× H F G = H ∘F (F ※ G)
+module Bifunctor (H : Bifunctor C D E) where
+  open Functor H public
 
-reduce-× : ∀ (H : Bifunctor D₁ D₂ C) (F : Functor E₁ D₁) (G : Functor E₂ D₂) → Bifunctor E₁ E₂ C
-reduce-× H F G = H ∘F (F ⁂ G)
+  overlap-× : ∀ (F : Functor A C) (G : Functor A D) → Functor A E
+  overlap-× F G = H ∘F (F ※ G)
 
-flip-bifunctor : Bifunctor C D E → Bifunctor D C E
-flip-bifunctor b = b ∘F Swap
+  module overlap-× {o‴ ℓ‴ e‴} {A : Category o‴ ℓ‴ e‴} (F : Functor A C) (G : Functor A D) = Functor (overlap-× F G)
 
-appˡ : Bifunctor C D E → Category.Obj C → Functor D E
-appˡ F c = F ∘F constˡ c
+  reduce-× : ∀ (F : Functor A C) (G : Functor B D) → Bifunctor A B E
+  reduce-× F G = H ∘F (F ⁂ G)
 
-appʳ : Bifunctor C D E → Category.Obj D → Functor C E
-appʳ F d = F ∘F constʳ d
+  flip : Bifunctor D C E
+  flip = H ∘F Swap
+
```

Comment:
It's unfortunate flip can't have a Bifunctor module, but modules don't support this kind of guarded corecursion.

Reply:
Now these aren't there anyway.